project/MimaBuild.scala

```diff
@@ -86,13 +86,13 @@ object MimaBuild {
 
   def mimaSettings(sparkHome: File, projectRef: ProjectRef): Seq[Setting[_]] = {
     val organization = "org.apache.spark"
-    val previousSparkVersion = "3.5.0"
+    val previousSparkVersion = "4.0.0"
     val project = projectRef.project
-    val id = "spark-" + project
+    val fullId = "spark-" + project + "_2.13"
```

Comment:
need to explicitly specify `_2.13`; otherwise, `_2.12` will be concatenated by default.